src/com/google/cose/Sign1Message.java

```diff
@@ -32,21 +32,30 @@
 public class Sign1Message extends CoseMessage {
   private final byte[] message;
   private final byte[] signature;
+  private final byte[] originalProtectedHeadersData;
 
-  Sign1Message(Map protectedHeaders, Map unprotectedHeaders, byte[] message, byte[] sign) {
+  Sign1Message(
+      Map protectedHeaders,
+      byte[] originalProtectedHeadersData,
```

Comment:
Would be nice to annotate this as `@Nullable` (unless the library is not using any annotations)

Reply:
Alternative idea:

`s/originalProtectedHeadersData/encodedProtectedHeaders`

Then, instead of passing null, you pass `CoseUtils.serializeProtectedHeaders(protectedHeaders)` to the ctor. This moves the branch into the builder, and it means a `Sign1Message` always has the encoded value. Though for this to work, I think the constructor would need to be private so that only the builder can call it.

Reply:
No existing use of annotations: https://github.com/search?q=repo%3Aandroid%2Fcose-lib%20annotations&type=code

Reply:
Done